context.go

```diff
@@ -126,6 +126,36 @@ func (c *Ctx) LoadProject(path string) (*Project, error) {
 	return p, nil
 }
 
+// resolveProjectRoot evaluates the root directory and does the following:
+//
+// If the passed path is a symlink outside GOPATH to a directory within a
+// GOPATH, the resolved full real path is returned.
+//
+// If the passed path is a symlink within a GOPATH, we return an error.
+//
+// If the passed path isn't a symlink at all, we just pass through.
+func (c *Ctx) resolveProjectRoot(path string) (string, error) {
+	// Determine if this is a symlink
```

Comment:
Initially I was thinking that you should `os.Lstat()` here and check the file mode to see if it's a symlink before going on to `filepath.EvalSymlinks()`, but on further review it seems like that might be superfluous. I'll leave it up to you - just wanting to make the note.